### PR TITLE
[MDS-5941] Create HSRC API

### PR DESCRIPTION
**Pull request by henryoforeh-dev**
## Objective 

[MDS-5941](https://bcmines.atlassian.net/browse/MDS-5941)

_Why are you making this change? Provide a short explanation and/or screenshots_

- Created migration script for audit columns in the mine report definition and compliance article reference table.
- Implemented save and update end points for compliance article resource.
- Implemented end points to create relationship between mine report definition and compliance article.

image

image

### CODE REVIEW

File: services/core-api/app/api/mines/reports/resources/mine_report_definition_compliance_article_xref_resource.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        mine_report_definition_id = data.get('mine_report_definition_id')
+        compliance_article_id = data.get('compliance_article_id')
+
+        update_mine_report_definition_compliance = MineReportDefinitionComplianceArticleXref \
+            .find_by_mine_report_definition_compliance_article_xref_guid(
+            mine_report_definition_compliance_article_xref_guid)
+
+        if update_mine_report_definition_compliance is None:
+            raise NotFound('Mine Report Definition Compliance Article Reference found')
```
Comment (taraepp) [inline services/core-api/app/api/mines/reports/resources/mine_report_definition_compliance_article_xref_resource.py]: I think this message is missing a "not"

Comment (henryoforeh-dev) [reply services/core-api/app/api/mines/reports/resources/mine_report_definition_compliance_article_xref_resource.py]: Thanks for catching that. I will update it.

File: services/core-api/app/api/compliance/resources/compliance_article_update_resource.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        compliance_article = ComplianceArticle()
+        try:
+            update_compliance_article_codes = compliance_article.update_all(compliance_article_codes)
+
+            if len(update_compliance_article_codes) == len(compliance_article_codes):
+                compliance_article.save_all(update_compliance_article_codes)
+
+            return update_compliance_article_codes
+        except MarshmallowError as e:
+            raise BadRequest(f'Something went wrong while trying to update {e}')
```
Comment (taraepp) [inline services/core-api/app/api/compliance/resources/compliance_article_update_resource.py]: Not sure about returning {e} in the error message if we're trying to hide SQL/system errors. But a console error would be handy for debugging if something goes wrong in prod.

File: services/core-api/app/api/mines/reports/resources/mine_report_definition_compliance_article_xref_resource.py (hunk 1 above)
Comment (henryoforeh-dev) [reply services/core-api/app/api/mines/reports/resources/mine_report_definition_compliance_article_xref_resource.py]: Updated

File: services/core-api/app/api/compliance/resources/compliance_article_update_resource.py (hunk 2 above)
Comment (henryoforeh-dev) [reply services/core-api/app/api/compliance/resources/compliance_article_update_resource.py]: Updated. I removed it and added it to logger error.

**Review by taraepp — APPROVED** [review]

**Review by isuru-aot — APPROVED** [review]

**Pull request merged**